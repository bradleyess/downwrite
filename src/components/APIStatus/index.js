import React from 'react'
import { Block } from 'glamor/jsxstyle'

const pos = {
  bottom: 8,
  right: 8,
  position: 'fixed'
}

export default ({ data, env }) => (
  <Block
    fontSize={12}
    opacity='.375'
    background='#CF4832'
    color='white'
    textAlign='center'
    padding={8}
    {...pos}>
    {data && (
      <span>
        {data.post} {env}
      </span>
    )}
  </Block>
)
