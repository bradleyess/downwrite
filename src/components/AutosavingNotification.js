import React from 'react'
import { Block } from 'glamor/jsxstyle'

/* TypeError: cyclic object value */
export default () => (
	<Block
		width={160}
		textAlign="center"
		top={20}
		right={0}
		left={0}
		zIndex="900"
		position="fixed"
		margin="auto"
		backgroundColor="white"
		boxShadow="0 0 2px rgba(0,0,0,.07), 0 2px 4px rgba(0,0,0,.12)"
		padding={4}>
		Autosaving
	</Block>
)
