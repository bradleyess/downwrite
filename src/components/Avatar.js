import React from 'react'
import { Block } from 'glamor/jsxstyle'

class RandomColor extends React.Component {
	state = {
		a: null,
		b: null
	}

	componentWillMount() {
		this.setState({
			a: `#FEB692`,
			b: `#EA5455`
		})
	}

	shouldComponentUpdate() {}

	render() {
		return this.props.children(this.state.a, this.state.b)
	}
}

export default () => (
	<RandomColor>
		{(A, B) => (
			<Block
				borderRadius={32}
				width={32}
				height={32}
				background={`linear-gradient(135deg, ${A} 10%, ${B} 100%)`}
				margin="auto"
			/>
		)}
	</RandomColor>
)
