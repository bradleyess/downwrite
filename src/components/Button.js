// @flow

import React from 'react'
import { css } from 'glamor'

const btn = css({
	display: 'block',
	fontSize: 14,
	backgroundColor: 'var(--color-6)',
	transition: 'background-color 250ms ease-in-out',
	color: '#282828',
	fontWeight: 700,
	border: 0,
	paddingTop: 4,
	paddingLeft: 18,
	paddingRight: 18,
	paddingBottom: 4,
	borderRadius: 4,
	boxShadow: '0 0 4px rgba(0,0,0,.037), 0 4px 8px rgba(0,0,0,.07)',
	':hover': {
		backgroundColor: 'var(--color-7)'
	}
})

type Props = {
	args: Object
}

const Button = ({ ...args }: Props) => <button className={css(btn)} {...args} />

export default Button
