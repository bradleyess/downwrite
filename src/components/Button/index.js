import React from 'react'
import { css } from 'glamor'

const BUTTON_SIZE = 62.5

const btn = css({
  display: 'block',
  backgroundColor: '#2584A5',
  transition: 'background-color 250ms ease-in-out',
  color: 'white',
  border: 0,
  padding: 16,
  borderRadius: '50%',
  maxHeight: BUTTON_SIZE,
  width: BUTTON_SIZE,
  height: '100%',
  boxShadow: '0 0 4px rgba(0,0,0,.07), 0 4px 8px rgba(0,0,0,.14)',
  ':hover': {
    backgroundColor: 'var(--color-2)'
  }
})

const positionedSty = css({
  position: 'absolute',
  left: -32,
  top: -32
})

const Button = ({ onClick, positioned, type, ...args }) => {
  let btnStyle = css(
    positioned ? [btn, positionedSty] : btn
  )

  return (
    <button onClick={onClick} className={btnStyle} {...args} />
  )
}

export default Button
