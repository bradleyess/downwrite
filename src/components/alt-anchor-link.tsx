import * as React from "react";
import classNames from "../utils/classnames";

interface IAltAnchorProps<T> extends React.AnchorHTMLAttributes<T> {
  space?: string;
}

export default React.forwardRef<HTMLAnchorElement, IAltAnchorProps<unknown>>(
  (props: IAltAnchorProps<unknown>, ref) => {
    const cx = classNames("AltLink", props.className);
    return (
      <a href={props.href} className={cx} style={props.style} ref={ref}>
        {props.children}
      </a>
    );
  }
);
