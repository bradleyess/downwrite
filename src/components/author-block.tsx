import * as React from "react";
import { Link } from "react-router-dom";
import Avatar from "./avatar";

interface IAuthorProps {
  authed: boolean;
  name: string;
  colors: string[];
}

export default function Author(props: IAuthorProps): JSX.Element {
  return (
    <aside className="Sheet AuthorBlock Wrapper Wrapper--sm">
      <header className="AuthorBlockHeader">
        <Avatar colors={props.colors} />
        <h6 className="AuthorBlockTitle">Post from {props.name}</h6>
      </header>
      {!props.authed && (
        <React.Fragment>
          <hr />
          <p className="AuthorBlockContent">
            <span>
              You can write and share on Downwrite, you can sign up or log in{" "}
            </span>
            <Link to="/login">here</Link>
          </p>
        </React.Fragment>
      )}
    </aside>
  );
}
