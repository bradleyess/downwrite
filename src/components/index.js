import Button from './Button'
import Card from './Card'
import DWEditor from './Draft'
import EmptyPosts from './EmptyPosts'
import Header from './Header'
import Input from './Input'
import Loading from './Loading'
import Logo from './Logo'
import Nav from './Nav'
import PostList from './PostList'
import Toolbar from './Toolbar'
import Wrapper from './Wrapper'
import Helpers from './Helpers'

export {
	Helpers,
	DWEditor,
	Wrapper,
	Toolbar,
	Button,
	Card,
	EmptyPosts,
	Header,
	Input,
	Loading,
	Logo,
	Nav,
	PostList
}
