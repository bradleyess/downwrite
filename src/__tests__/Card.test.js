import { Card } from '../components'
import { MemoryRouter } from 'react-router-dom'
import { posts } from './db.json'

const mockDelete = jest.fn()
const id = '6acebce0-20b6-4015-87fe-951c7bb36481'

const wrapper = mount(
	<MemoryRouter>
		<Card title='Starting Again' content={posts[0].content} id={id} onDelete={mockDelete} />
	</MemoryRouter>
)

describe('Card', () => {
	it('contains snippet from content', () => {
		const snippet = wrapper.find('[data-test="cardDelete"]').text()
		expect(snippet.length).toBeLessThanOrEqual(90)
	})

	it('contains a title', () => {
		expect(wrapper.find('h2').text()).toBe('Starting Again')
	})


	it('should fire a delete', () => {
		wrapper.find('[data-test="cardDelete"]').simulate('click')

		expect(mockDelete).toHaveBeenCalled()
	})
})
