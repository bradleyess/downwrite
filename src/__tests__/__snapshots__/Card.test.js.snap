// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
.glamor-0,
[data-glamor-0] {
  font-size: 14px;
  font-weight: 500;
  font-family: var(--secondary-font);
}

.glamor-8,
[data-glamor-8] {
  font-size: small;
  opacity: 0.875;
  padding: 16px;
}

.glamor-9,
[data-glamor-9] {
  border-left-width: 4px;
  border-left-style: solid;
  border-color: var(--color-1);
  height: 100%;
  font-weight: 500;
  box-shadow: 0 0 2px rgba(0,0,0,.07), 0 2px 4px rgba(0,0,0,.12);
  background-color: white;
}

.glamor-1,
[data-glamor-1] {
  opacity: 0.5;
  font-family: var(--primary-font);
  font-weight: 400;
}

.glamor-2,
[data-glamor-2] {
  font-size: 12px;
  text-align: right;
  text-transform: uppercase;
  letter-spacing: 2px;
}

.css-fisw11,
[data-css-fisw11] {
  display: block;
}

.css-fo8gak,
[data-css-fo8gak] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  border-bottom: 1px solid #f2f2f2;
  padding: 16px;
  justify-content: space-between;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-box-align: start;
  -webkit-align-items: flex-start;
}

.css-1t62idy,
[data-css-1t62idy] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.glamor-4,
[data-glamor-4] {
  font-size: 12px;
  text-align: right;
  text-transform: uppercase;
  letter-spacing: 2px;
  color: var(--color-2);
  border: 0;
  appearance: none;
  -webkit-font-smoothing: antialiased;
  -webkit-appearance: none;
  -moz-appearance: none;
}

@media (min-width: 57.75rem) {
  .glamor-0,
  [data-glamor-0] {
    font-size: 16px;
  }
}

@media (min-width: 57.75rem) {
  .glamor-2,
  [data-glamor-2] {
    font-size: 14px;
    font-weight: 700;
  }
}

@media (min-width: 57.75rem) {
  .glamor-4,
  [data-glamor-4] {
    font-size: 14px;
    font-weight: 700;
  }
}

<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "msfz54",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "msfz54",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Card
      content={
        Object {
          "blocks": Array [
            Object {
              "data": Object {},
              "depth": 0,
              "entityRanges": Array [],
              "inlineStyleRanges": Array [],
              "key": "er5rb",
              "text": "What am i really going to do here?",
              "type": "unstyled",
            },
            Object {
              "data": Object {},
              "depth": 0,
              "entityRanges": Array [],
              "inlineStyleRanges": Array [],
              "key": "ds3fg",
              "text": "",
              "type": "unstyled",
            },
            Object {
              "data": Object {},
              "depth": 0,
              "entityRanges": Array [],
              "inlineStyleRanges": Array [],
              "key": "9sgm6",
              "text": "I am trying to write a thing",
              "type": "unstyled",
            },
            Object {
              "data": Object {},
              "depth": 0,
              "entityRanges": Array [],
              "inlineStyleRanges": Array [],
              "key": "a0t8m",
              "text": "",
              "type": "unstyled",
            },
            Object {
              "data": Object {},
              "depth": 0,
              "entityRanges": Array [],
              "inlineStyleRanges": Array [],
              "key": "2ok0",
              "text": "With many lines",
              "type": "unstyled",
            },
          ],
          "entityMap": Object {},
        }
      }
      id="6acebce0-20b6-4015-87fe-951c7bb36481"
      onDelete={[Function]}
      title="Starting Again"
    >
      <Block
        className={
          Object {
            "data-glamor-9": "",
          }
        }
      >
        <View
          className={
            Object {
              "data-glamor-9": "",
            }
          }
          component="div"
          display="block"
        >
          <div
            className={
              Object {
                "data-glamor-9": "",
              }
            }
            data-css-fisw11=""
          >
            <Flex
              alignItems="flex-start"
              borderBottom="1px solid #f2f2f2"
              justifyContent="space-between"
              padding={16}
            >
              <View
                alignItems="flex-start"
                borderBottom="1px solid #f2f2f2"
                component="div"
                display="flex"
                justifyContent="space-between"
                padding={16}
              >
                <div
                  data-css-fo8gak=""
                >
                  <div>
                    <h2
                      className={
                        Object {
                          "data-glamor-0": "",
                        }
                      }
                    >
                      <Link
                        replace={false}
                        to="/6acebce0-20b6-4015-87fe-951c7bb36481/edit"
                      >
                        <a
                          href="/6acebce0-20b6-4015-87fe-951c7bb36481/edit"
                          onClick={[Function]}
                        >
                          Starting Again
                        </a>
                      </Link>
                    </h2>
                    <small
                      className={
                        Object {
                          "data-glamor-1": "",
                        }
                      }
                    />
                  </div>
                  <Flex
                    className={Object {}}
                    flexDirection="column"
                  >
                    <View
                      className={Object {}}
                      component="div"
                      display="flex"
                      flexDirection="column"
                    >
                      <div
                        className={Object {}}
                        data-css-1t62idy=""
                      >
                        <EditButton
                          id="6acebce0-20b6-4015-87fe-951c7bb36481"
                        >
                          <Link
                            className={
                              Object {
                                "data-glamor-2": "",
                              }
                            }
                            replace={false}
                            to="/6acebce0-20b6-4015-87fe-951c7bb36481/edit"
                          >
                            <a
                              className={
                                Object {
                                  "data-glamor-2": "",
                                }
                              }
                              href="/6acebce0-20b6-4015-87fe-951c7bb36481/edit"
                              onClick={[Function]}
                            >
                              <svg
                                height={18}
                                viewBox="0 0 18 18"
                                width={18}
                              >
                                <desc>
                                  Edit this Post
                                </desc>
                                <g
                                  fill="none"
                                  fillRule="evenodd"
                                  id="Views"
                                  stroke="none"
                                  strokeWidth="1"
                                >
                                  <g
                                    id="Artboard"
                                    transform="translate(-70.000000, -134.000000)"
                                  >
                                    <g
                                      id="ic_mode_edit_black_24px"
                                      transform="translate(67.000000, 131.000000)"
                                    >
                                      <path
                                        d="M3,17.25 L3,21 L6.75,21 L17.81,9.94 L14.06,6.19 L3,17.25 Z M20.71,7.04 C21.1,6.65 21.1,6.02 20.71,5.63 L18.37,3.29 C17.98,2.9 17.35,2.9 16.96,3.29 L15.13,5.12 L18.88,8.87 L20.71,7.04 Z"
                                        fill="#185A70"
                                        fillRule="nonzero"
                                        id="Shape"
                                      />
                                      <polygon
                                        id="Shape"
                                        points="0 0 24 0 24 24 0 24"
                                      />
                                    </g>
                                  </g>
                                </g>
                              </svg>
                            </a>
                          </Link>
                        </EditButton>
                        <DeleteButton
                          onDelete={[Function]}
                        >
                          <span
                            className={
                              Object {
                                "data-glamor-4": "",
                              }
                            }
                            data-test="cardDelete"
                            onClick={[Function]}
                          >
                            <svg
                              height={18}
                              viewBox="0 0 14 18"
                              width={18}
                            >
                              <desc>
                                Delete this Post
                              </desc>
                              <g
                                fill="none"
                                fillRule="evenodd"
                                id="Views"
                                stroke="none"
                                strokeWidth="1"
                              >
                                <g
                                  id="Artboard"
                                  transform="translate(-72.000000, -172.000000)"
                                >
                                  <g
                                    id="ic_delete_forever_black_24px"
                                    transform="translate(67.000000, 169.000000)"
                                  >
                                    <polygon
                                      id="Shape"
                                      points="0 0 24 0 24 24 0 24"
                                    />
                                    <path
                                      d="M6,19 C6,20.1 6.9,21 8,21 L16,21 C17.1,21 18,20.1 18,19 L18,7 L6,7 L6,19 Z M8.46,11.88 L9.87,10.47 L12,12.59 L14.12,10.47 L15.53,11.88 L13.41,14 L15.53,16.12 L14.12,17.53 L12,15.41 L9.88,17.53 L8.47,16.12 L10.59,14 L8.46,11.88 Z M15.5,4 L14.5,3 L9.5,3 L8.5,4 L5,4 L5,6 L19,6 L19,4 L15.5,4 Z"
                                      fill="#D8EAF1"
                                      fillRule="nonzero"
                                      id="Shape"
                                    />
                                    <polygon
                                      id="Shape"
                                      points="0 0 24 0 24 24 0 24"
                                    />
                                  </g>
                                </g>
                              </g>
                            </svg>
                          </span>
                        </DeleteButton>
                      </div>
                    </View>
                  </Flex>
                </div>
              </View>
            </Flex>
            <Block>
              <View
                component="div"
                display="block"
              >
                <div
                  data-css-fisw11=""
                >
                  <p
                    className={
                      Object {
                        "data-glamor-8": "",
                      }
                    }
                    data-test="snippet"
                  >
                    What am i really going to do here?
                  </p>
                </div>
              </View>
            </Block>
          </div>
        </View>
      </Block>
    </Card>
  </Router>
</MemoryRouter>
`;
